src/Microsoft.Net.Http.Headers/NameValueHeaderValue.cs

```diff
@@ -390,13 +413,109 @@ private static void CheckValueFormat(StringSegment value)
             // Either value is null/empty or a valid token/quoted string
             if (!(StringSegment.IsNullOrEmpty(value) || (GetValueLength(value, 0) == value.Length)))
             {
-                throw new FormatException(string.Format(System.Globalization.CultureInfo.InvariantCulture, "The header value is invalid: '{0}'", value));
+                throw new FormatException(string.Format(CultureInfo.InvariantCulture, "The header value is invalid: '{0}'", value));
             }
         }
 
         private static NameValueHeaderValue CreateNameValue()
         {
             return new NameValueHeaderValue();
         }
+
+        // See https://tools.ietf.org/html/rfc7230#section-3.2.6
+        private static StringSegment UnescapeAsQuotedString(StringSegment input)
+        {
+            input = HeaderUtilities.RemoveQuotes(input);
+
+            // First pass to calculate the size of the InplaceStringBuilder
+            var backSlashCount = CountBackslashesForDecodingQuotedString(input);
+
+            if (backSlashCount == 0)
+            {
+                return input;
+            }
+
+            var stringBuilder = new InplaceStringBuilder(input.Length - backSlashCount);
+
+            for (var i = 0; i < input.Length; i++)
+            {
+                if (input[i] == '\\')
+                {
+                    if (input.Length == i + 1)
+                    {
+                        throw new FormatException(string.Format(CultureInfo.InvariantCulture,
+                            "The header value is invalid, illegal escape character in input: '{0}', position: '{1}'", input, i + 1));
```

Comment:
Change the first comma to a period, a colon, or a semicolon; the comma is incorrect.